compiler/circle-execution-plan/src/ExecutionPlanner.cpp

```diff
@@ -116,9 +82,75 @@ uint32_t compute_im2col_size(const luci::CircleConv2D *conv)
 
 } // namespace
 
+void ExecutionPlanner::make_rpo_execution_plan()
+{
+  _ordered_nodes = loco::postorder_traversal(loco::output_nodes(const_cast<loco::Graph *>(_graph)));
+  make_execution_plan();
+}
+
+void ExecutionPlanner::make_exhaustive_search_execution_plan()
+{
+  std::vector<loco::Node *> best_execution_order;
+  uint32_t best_required_size = std::numeric_limits<uint32_t>::max();
+  std::vector<NodeOrderInform> node_order_inform_vector;
+
+  ExecutionOrderEstimator execution_order_estimator(
+    loco::output_nodes(const_cast<loco::Graph *>(_graph)));
+  execution_order_estimator.get_node_order_inform_vector(node_order_inform_vector);
+
+  auto start_pos = static_cast<int32_t>(node_order_inform_vector.size() - 1);
+  int32_t end_pos = 0;
+  int32_t current_pos = start_pos;
+
+  while (current_pos >= end_pos)
+  {
+    if (current_pos == start_pos)
+    {
+      while (node_order_inform_vector.at(current_pos).current_idx <
+             node_order_inform_vector.at(current_pos).all_combinations.size())
+      {
+        _ordered_nodes = execution_order_estimator.get_current_order(node_order_inform_vector);
+
+        get_usage_interval();
+        _required_size = greedy_by_size_approach();
+
+        if (_required_size < best_required_size)
+        {
+          best_required_size = _required_size;
+          best_execution_order = _ordered_nodes;
+        }
+
+        _ordered_nodes.clear();
+        _alloc_node_inform_vector.clear();
+        _alloc_node.clear();
+        _dealloc_node.clear();
```

Comment:
optional: Move this to separate method, something like "clean_state", so we can call it from other places to reset planning state during some search algorithm.

Alternative way is to move memory planning stuff (including `make_execution_plan` method) into separate class.